### PR TITLE
Use a single SplinterProtocolVersion value across libsplinter

### PR DESCRIPTION
**Pull request by vaporos**
This replaces crate::rest_api::*_PROTOCOL_VERSION with a single
SPLINTER_PROTOCOL_VERSION. This is consistent with the original
intended design and the future direction of the entire Splinter REST API
being versioned as a whole (as opposed to versioned per-component).

### CODE REVIEW

**Review by peterschwarz — APPROVED** [review]

File: libsplinter/src/protocol/mod.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ pub mod service;
 
 // Peer authorization protocol versions
 #[cfg(any(feature = "trust-authorization", feature = "challenge-authorization"))]
-pub const PEER_AUTHORIZATION_PROTOCOL_VERSION: u32 = 1;
+pub const PEER_SPLINTER_PROTOCOL_VERSION: u32 = 1;
```
Comment (agunde406) [inline libsplinter/src/protocol/mod.rs:24]: This changes should be split into a different commit.

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: libsplinter/src/protocol/mod.rs (hunk 1 above)
Comment (vaporos) [reply libsplinter/src/protocol/mod.rs]: This was unintentional. Removed from the commit.

**Review by agunde406 — APPROVED** [review]

**Review by isabeltomb — APPROVED** [review]
